#392 update contribution guidelines

nilsnolde opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box when you have checked you have done them): -->
- [x] 1. I have merged the latest version of the development branch into my feature branch and all conflicts have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the app.config file, I have added these to the app.config.sample file along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes # .

### Information about the changes
- Reason for change: 

Improved contribution guidelines. Awaiting #393 to link in doc. Had to make it 2 commits since it involves renaming a file..

Commit:
rename contributing.md

Commit:
update CONTRIBUTE.md with drafted guidelines

Commit:
add git stash to rebase instructions [ci skip]

rabidllama commented on CONTRIBUTE.md:
```diff
@@ -0,0 +1,89 @@
+
+Be extra careful using `rebase` commands when collaborating on a branch with other people.
+
+**Don't merge** `development` branch into your feature branch.
```
I'd suggest we update the PR request template then as it says in there that "I have merged development into my branch". This should be done in this PR so that things stay clean :p

nilsnolde replied:
> I'd suggest we update the PR request template then as it says in there that "I have merged development into my branch".

Well, we don't want anyone to **merge** the head branch back to their feature branch. That's what this whole section is about. No merge, but rebase, right?

> This should be done in this PR so that things stay clean :p

What does that mean? You wanna drop these 2 comments and include them in the template?

nilsnolde replied:
Ah forget it, I see what you mean. I'll update the PR template!

nilsnolde replied:
done

rabidllama commented on CONTRIBUTE.md:
```diff
@@ -0,0 +1,89 @@
+We'd like all pull requests to adhere to the following rules:
+
+- a PR has to **close an issue**. If there is none yet for yours, please [create one](https://github.com/GIScience/openrouteservice/issues/new)
+- branch off [developemnt](https://github.com/GIScience/openrouteservice/tree/development)
```
Spelling - development

nilsnolde replied:
fixed

rabidllama replied:
but have you? :smirk: still says developemnt

nilsnolde replied:
uff, right..

rabidllama reviewed (changes requested):
Couple of minor things that should be updated before the PR is finished

Commit:
fix PR template to rebase, fix typo

Commit:
fixed typo for real;)

Commit:
adapt README links

nilsnolde commented:
Please wait with merging until I created a quick setup guide (#393)

nilsnolde committed:
add ../ to data paths

so we can just copy the config to WEB-INF and use it without changing paths manually

nilsnolde commented:
Ok, PR is ready. Changed README.md to reflect CONTRIBUTING.md and added `../` to data paths in `app.config.test` to better use it out-of-the-box

nilsnolde committed:
update wiki entry for code contribution

nilsnolde committed:
revert changing data paths

Commit:
Merge branch 'development' into 392-contribution-md
